library/std/src/thread/local.rs

```diff
@@ -205,7 +209,13 @@ macro_rules! __thread_local_inner {
                 // just get going.
                 if !$crate::mem::needs_drop::<$t>() {
                     unsafe {
-                        return Some(&VAL)
+                        // FIXME: remove the #[allow(...)] marker when macros don't
+                        // raise warning for missing/extraneous unsafe blocks anymore.
+                        // See https://github.com/rust-lang/rust/issues/74838.
+                        #[allow(unused_unsafe)]
+                        unsafe {
+                            return Some(&VAL);
+                        }
                     }
```

Comment:
Why is it necessary to put an `unsafe {}` inside another `unsafe {}` block?

Reply:
I don't think it is. Good catch!